src/test/java/uk/gov/justice/hmpps/casenotes/services/CaseNoteTypeMergerTest.java

```diff
@@ -49,7 +49,45 @@ public void testSimpleMerge() {
                         ).build()
                 ).toArray(new CaseNoteType[0])
         );
+    }
+
+    @Test
+    public void testActiveFlagUpdatedWhenParentInactiveDuringMerge() {
+
+        final var list1 = List.of(
+                CaseNoteType.builder().code("OBS").description("Observation").activeFlag("Y").subCodes(
+                        List.of(CaseNoteType.builder().code("GEN").description("General").activeFlag("Y").build(), CaseNoteType.builder().code("SP").description("Special").activeFlag("N").build())
+                ).build(),
+                CaseNoteType.builder().code("DRR").description("Drug Rehabilitation Requirement").activeFlag("N").subCodes(
+                        List.of(CaseNoteType.builder().code("DCOUN").description("Drug Counselling Session").activeFlag("Y").build(), CaseNoteType.builder().code("DTEST").description("Drug Test").activeFlag("N").build())
+                ).build()
+        );
 
+        final var list2 = List.of(
+                CaseNoteType.builder().code("OBS").description("Observation").activeFlag("Y").subCodes(
+                        List.of(CaseNoteType.builder().code("NEW").description("New Stuff").activeFlag("Y").build(), CaseNoteType.builder().code("GEN").activeFlag("Y").description("Different Gen").activeFlag("Y").build())
+                ).build(),
+                CaseNoteType.builder().code("POM").description("POM Stuff").activeFlag("Y").subCodes(
+                        List.of(CaseNoteType.builder().code("SPC").description("Special").activeFlag("Y").build(), CaseNoteType.builder().code("GEN").description("General").activeFlag("Y").build())
+                ).build()
+        );
 
+        List<CaseNoteType> resultantList = merger.mergeAndSortList(list1, list2);
+
+        assertThat(resultantList).hasSize(3);
+
+        assertThat(resultantList).containsExactly(
+                List.of(
+                        CaseNoteType.builder().code("DRR").description("Drug Rehabilitation Requirement").activeFlag("N").subCodes(
```

Comment:
not really clear what you're testing against, just seems to be lots of builders going on

Reply:
maybe if you add a `createInactive` to do an inactive build and a `createActive` to do an active build then it would be lots clearer.  Would be good to have scenario where nomis is inactive and ocn is active, seems to be missing.  Also worth adding test when there is only a `list1` which will show scenario.

Reply:
alternatively looks much nicer in kotlin
![Screenshot from 2019-09-11 08-02-38](https://user-images.githubusercontent.com/5099053/64675275-bc29f900-d46a-11e9-9fdb-7e718ac8272b.png)

Reply:
@JamieDraperUK :point_up: